Sources/Memcache/ChannelHandler/MemcacheBackendMessageDecoder.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import NIOCore
+
+struct MemcacheBackendMessageDecoder: NIOSingleStepByteToMessageDecoder {
+    typealias InboundOut = MemcacheBackendMessage
+
+    func decode(buffer: inout ByteBuffer) throws -> MemcacheBackendMessage? {
+        // Keep track of the reader index in case we later notice that we need more data
+        let startReaderIndex = buffer.readerIndex
+
+        // Peek at the message to read the verb. It is before the first \r\n and before the first <space> if the message
+        // contains one.
+        guard let messageSlice = buffer.getCarriageReturnNewlineTerminatedSlice(at: buffer.readerIndex) else {
+            // reader index wasn't moved, wait for more bytes
+            return nil
+        }
```

Comment:
What is the benefit of finding the first line, if that doesn't ensure that this will be the complete first message? IIUC you are interested in the first `VERB` and you can get that by finding either a space or a `\r\n` correct?